irohad/main/impl/pg_connection_init.cpp

```diff
@@ -0,0 +1,302 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "main/impl/pg_connection_init.hpp"
+
+#include "logger/logger.hpp"
+#include "logger/logger_manager.hpp"
+
+using namespace iroha::ametsuchi;
+
+std::string PgConnectionInit::formatPostgresMessage(const char *message) {
+  std::string formatted_message(message);
+  boost::replace_if(formatted_message, boost::is_any_of("\r\n"), ' ');
+  return formatted_message;
+}
+
+void PgConnectionInit::processPqNotice(void *arg, const char *message) {
+  auto *log = reinterpret_cast<logger::Logger *>(arg);
+  log->debug("{}", formatPostgresMessage(message));
+}
+
+iroha::expected::Result<std::shared_ptr<soci::connection_pool>, std::string>
+PgConnectionInit::initPostgresConnection(std::string &options_str,
+                                         size_t pool_size) {
+  auto pool = std::make_shared<soci::connection_pool>(pool_size);
+
+  try {
+    for (size_t i = 0; i != pool_size; i++) {
+      soci::session &session = pool->at(i);
+      session.open(*soci::factory_postgresql(), options_str);
+    }
+  } catch (const std::exception &e) {
+    return expected::makeError(formatPostgresMessage(e.what()));
+  }
+  return expected::makeValue(pool);
+}
+
+iroha::expected::Result<std::unique_ptr<PoolWrapper>, std::string>
+PgConnectionInit::prepareConnectionPool(
+    ReconnectionStrategyFactory &reconnection_strategy_factory,
+    const PostgresOptions &options,
+    logger::LoggerManagerTreePtr log_manager) {
+  const int pool_size = 10;
```

Comment:
I would suggest to make it a parameter as it was before, and define this variable in `Irohad::initStorage`